source/particle/property/interface.cc

```diff
@@ -130,6 +140,49 @@ namespace aspect
 
       template <int dim>
       void
+      Manager<dim>::initialize_late_particle (Particle<dim> &particle,
+                                              const std::multimap<types::LevelInd, Particle<dim> > &particles,
+                                              const Interpolator::Interface<dim> &interpolator,
+                                              const Vector<double> &solution,
+                                              const std::vector<Tensor<1,dim> > &gradients) const
+      {
+        std::vector<double> particle_properties (0);
+
+        unsigned int property_index = 0;
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p, ++property_index)
+          {
+            if ((*p)->late_initialize_mode() == InitializeLateParticles::initialize)
```

Comment:
I think a better structure would replace the repeated `if - if else - if else` statements by a `switch -case -case` approach.